frame/uniswap-v2/src/lib.rs

```diff
@@ -0,0 +1,863 @@
+		ExternalAssetCheckFailed,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Pool with specified id `PoolId` was created successfully by `T::AccountId`.
+		///
+		/// Included values are:
+		/// - account identifier `T::AccountId`
+		/// - pool identifier `PoolId`
+		///
+		/// \[who, pool_id\]
+		PoolCreated { who: T::AccountId, pool_id: PoolId },
+
+		/// Liquidity added into the pool `PoolId` by `T::AccountId`.
+		///
+		/// Included values are:
+		/// - account identifier `T::AccountId`
+		/// - pool identifier `PoolId`
+		/// - added token amounts `Vec<T::Balance>`
+		/// - charged fees `Vec<T::Balance>`
+		/// - actual invariant `T::Balance`
+		/// - actual token supply `T::Balance`
+		/// - minted amount `T::Balance`
+		///
+		/// \[who, pool_id, token_amounts, fees, invariant, token_supply, mint_amount\]
+		LiquidityAdded {
+			who: T::AccountId,
+			pool_id: PoolId,
+			token_amounts: Vec<T::Balance>,
+			fees: Vec<T::Balance>,
+			invariant: T::Balance,
+			token_supply: T::Balance,
+			mint_amount: T::Balance,
+		},
+
+		/// Liquidity removed from pool `PoolId` by `T::AccountId` in balanced way.
+		///
+		/// Included values are:
+		/// - account identifier `T::AccountId`
+		/// - pool identifier `PoolId`
+		/// - removed token amounts `Vec<T::Balance>`
+		/// - charged fees `Vec<T::Balance>`
+		/// - actual token supply `T::Balance`
+		///
+		/// \[who, pool_id, token_amounts, fees, token_supply\]
+		LiquidityRemoved {
+			who: T::AccountId,
+			pool_id: PoolId,
+			token_amounts: Vec<T::Balance>,
+			fees: Vec<T::Balance>,
+			token_supply: T::Balance,
+		},
+
+		/// Token exchange happened.
+		///
+		/// Included values are:
+		/// - account identifier `T::AccountId`
+		/// - pool identifier `PoolId`
+		/// - index of sent token `PoolTokenIndex`
+		/// - amount of sent token `T::Balance`
+		/// - index of received token `PoolTokenIndex`
+		/// - amount of received token `T::Balance`
+		/// - charged fee `T::Balance`
+		///
+		/// \[who, pool_id, sent_token_index, sent_amount, received_token_index, received_amount,
+		/// fee\]
+		TokenExchanged {
+			who: T::AccountId,
+			pool_id: PoolId,
+			sent_token_index: PoolTokenIndex,
+			sent_amount: T::Balance,
+			received_token_index: PoolTokenIndex,
+			received_amount: T::Balance,
+			fee: T::Balance,
+		},
+
+		/// Withdraw admin fees `Vec<T::Balance>` from pool `PoolId` by user `T::AccountId`
+		///
+		/// Included values are:
+		/// - account identifier `T::AccountId`
+		/// - pool identifier `PoolId`
+		/// - admin fee receiving account identifier `T::AccountId`
+		/// - withdrew admin fees `Vec<T::Balance>`
+		///
+		/// [who, pool_id, admin_fee_account, admin_fees]
+		AdminFeesWithdrawn {
+			who: T::AccountId,
+			pool_id: PoolId,
+			admin_fee_account: T::AccountId,
+			admin_fees: Vec<T::Balance>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+
+	impl<T: Config> CurveAmm for Pallet<T> {
```

Comment:
please split curves code

Reply:
Don't get this comment. Both models are curve. So CurveAmm is generic traits for DEX based on curve. Although I have planned few refactoring in next PR.